assembl/scripts/ini_files.py

```diff
@@ -144,72 +166,392 @@ def main():
+    if overlay._defaults:
+        config._defaults.update(overlay._defaults)
+    return config
+
+
+def migrate_beaker_config(random_ini, overlay):
+    """Migrate old-style session... to beaker.session..."""
+    for section in random_ini.sections():
+        # Avoid including DEFAULTSECT
+        if not overlay.has_section(section):
+            continue
+        o_section = overlay._sections[section]
+        for key in random_ini._sections[section]:
+            if key.startswith("beaker."):
+                old_key = key[7:]
+                if old_key in o_section and key not in o_section:
+                    overlay.set(section, key, overlay.get(section, old_key))
+            elif key == 'session.secret':
+                alt_key = 'beaker.' + key
+                if alt_key in o_section and key not in o_section:
+                    overlay.set(section, key, overlay.get(section, alt_key))
+    return overlay
+
+
+def dump(ini_file):
+    """Dump the ini file, showing interpolations and errors."""
+    ini_file = asParser(ini_file, SafeConfigParser)
+    ini_file._defaults['here'] = os.getcwd()
+    for section in ini_file.sections():
+        print "\n[%s]\n" % section
+        for option in ini_file._sections[section]:
+            print option, '=',
+            try:
+                print ini_file.get(section, option)
+            except Exception as e:
+                print ini_file.get(section, option, True)
+                print "*"*15, "Could not interpolate"
+                print e
+
+
+def populate_random(random_file, random_templates=None, saml_info=None):
+    """Populate random.ini
+
+    Create missing random values according to the template
+    Do not change existing values"""
+    from base64 import b64encode
+    from os import urandom
+    from assembl.auth.make_saml import make_saml_key, make_saml_cert
+    base = Parser()
+    assert random_templates, "Please give one or more templates"
+    for template in random_templates:
+        assert exists(template), "Cannot find template " + template
+        base.read(template)
+    existing = Parser()
+    if exists(random_file):
+        existing.read(random_file)
+    combine_ini(base, existing)
+    saml_keys = {}
+    changed = False
+
+    for section in base.sections():
+        for key, value in base.items(section):
+            keyu = key.upper()
+            # too much knowdledge, but hard to avoid
+            if "SAML" in keyu and keyu.endswith("_PRIVATE_KEY"):
+                prefix = keyu[:-12]
+                if value == "{saml_key}":
+                    saml_key_text, saml_key = make_saml_key()
+                    base.set(section, key, saml_key_text)
+                    saml_keys[prefix] = saml_key
+                    changed = True
+                else:
+                    saml_keys[prefix] = value
+            elif value.startswith('{random') and value.endswith("}"):
+                value = b64encode(urandom(int(value[7:-1])))
```

Comment:
Added the size check. I think regexp is overkill when startswith/endswith is enough.